radix-engine-tests/tests/db/substate_database_overlay.rs

```diff
@@ -442,54 +464,89 @@ fn create_database_contents_hash<D: SubstateDatabase + ListableSubstateDatabase>
     accumulator_hash
 }
 
-/// Runs the scenarios on an [`InMemorySubstateDatabase`] and a [`SingleSubstateDatabaseOverlay`] wrapping
+trait DatabaseComparisonScenarioCheck {
+    fn check(
+        &mut self,
```

Comment:
Is self needed?

Reply:
Possibly not - this is just a test helper for two tests (see other comment)

Might just rewrite the tests if you’d think it were clearer?